src/configCreator.ts

```diff
@@ -0,0 +1,68 @@
+import { CLIEngine, Linter } from "eslint"
+import { defaultOptions } from "./eslintDefaultOptions"
+import { Codacyrc, Pattern, ParameterValue } from "./model/CodacyInput"
+import { toolName } from "./toolMetadata"
+import { patternIdToEslint } from "./model/Patterns"
+import { cloneDeep, isEmpty, partition, fromPairs } from "lodash"
+import { allFilesNames } from "./allFiles"
+
+function patternsToRules(
+  patterns: Pattern[]
+): { [name: string]: Linter.RuleLevel | Linter.RuleLevelAndOptions } {
+  const pairs = patterns.map(pattern => {
+    const patternId = patternIdToEslint(pattern.patternId)
+    if (pattern.parameters) {
+      const [unnamedParameters, namedParameters] = partition(
```

Comment:
you are doing a bunch of operations on `namedParameters` does it make sense to refactor in a class?

Reply:
I moved them to a separate file and I added a test for them.